pico/services/src/main/java/io/helidon/pico/services/State.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.pico.services;
+
+import java.util.Objects;
+
+import io.helidon.pico.Phase;
+import io.helidon.pico.Resettable;
+
+class State implements Resettable, Cloneable {
```

Comment:
Is thi salways used from a single thread?

Reply:
Not necessarily, no.